### PR TITLE
Upgrade oktthp3 to 3.14.9, Close connection after link activation req

### LINKED ISSUE #140: Kinan sometimes connects to the proxy without authentication
This only happens occasionally from what I can see. This causes issues since some proxy returns garbage without authentication to hide itself from probes. Is it possible to fix?

### PR DESCRIPTION
**Pull request by exiva**
Upgrading http library to 3.14.9 fixes #140 on kinancity-core, and closing the connection pool on linkacitvator in kinancity-mail fixes #140

### CODE REVIEW

File: KinanCity-mail/src/main/java/com/kinancity/mail/activator/DirectLinkActivator.java [hunk 1]
```diff
@@ -30,7 +32,11 @@ public class DirectLinkActivator implements LinkActivator {
 	private okhttp3.OkHttpClient client;
 
 	public DirectLinkActivator() {
-		client = new OkHttpClient.Builder().build();
+		new OkHttpClient.Builder()
+		.readTimeout(10,TimeUnit.SECONDS)
+		.retryOnConnectionFailure(false)
+		.connectionPool(new ConnectionPool(0,0,TimeUnit.SECONDS))
```
Comment (drallieiv) [inline KinanCity-mail/src/main/java/com/kinancity/mail/activator/DirectLinkActivator.java:38]: why force the pool to an emtpy one ?

Comment (Mygod) [reply KinanCity-mail/src/main/java/com/kinancity/mail/activator/DirectLinkActivator.java:38]: It's to ensure that it doesn't reuse the connection for other sessions. It's helpful for rotating proxies.

**Comment by clburlison** [comment]
Any blockers on this PR? It's been working successfully for me for the past few months.

**Comment by Mygod** [comment]
Should we add closing idle connections to core as well? (This PR seems to only add it to mail.) I am still seeing idle connections being kept around for 2-5 minutes before closing on core.

**Pull request merged**